### PR TITLE
feat: Add id prop to Text, TouchableWithoutFeedback and View docs

### PR DESCRIPTION
**Pull request by gabrieldonadel**
This PR updates `Text`, `TouchableWithoutFeedback`, and `View` documentations adding information about the new `id` prop.

Related to https://github.com/facebook/react-native/pull/34522

### CODE REVIEW

**Comment by gabrieldonadel** [comment]
@Simek FYI https://github.com/facebook/react-native/pull/34522 was merged a couple days ago

File: docs/text.md [hunk 1]
```diff
@@ -417,6 +417,16 @@ This can be one of the following values:
 
 ---
 
+### `id`
+
+Used to locate this view from native code. Has precedence over `nativeID`
```
Comment (Simek) [inline docs/text.md:422]:
```suggestion
Used to locate this view from native code. Has precedence over `nativeID` prop.
```

File: docs/view.md [hunk 2]
```diff
@@ -282,6 +282,18 @@ For example, if a touchable view has a height of 20 the touchable height can be
 
 ---
 
+### `id`
+
+Used to locate this view from native classes. Has precedence over `nativeID`
+
+> This disables the 'layout-only view removal' optimization for this view!
```
Comment (Simek) [inline docs/view.md:289]:
```suggestion
> This disables the 'layout-only view removal' optimization for this view!
```

Can you explain this part a bit broader or provide a reference (link) for the optimization you are referring to?

**Review by Simek — CHANGES_REQUESTED** [review]
See comments above.

File: docs/view.md (hunk 2 above)
Comment (gabrieldonadel) [reply docs/view.md:289]: I'm not 100% sure what this is TBH, but this is the exact same wording used in the `nativeID` section

https://github.com/facebook/react-native-website/blob/main/docs/view.md?plain=1#L309

File: docs/text.md (hunk 1 above)
Comment (gabrieldonadel) [reply docs/text.md]: Sure, I've just updated it

File: docs/view.md (hunk 2 above)
Comment (Simek) [reply docs/view.md:289]: Oh, I see so you have just copied it over from the other prop. Let's leave this as-is for now, and I will ask around if someone can clarify this not a bit (or maybe even it's no longer valid, will see).

**Review by Simek — APPROVED** [review]
Thank you for the review updates and clarification! 🙏

**Pull request merged**